File: app/src/main/java/com/example/kr_kotlin/ui/registration_java/RegistrFragment.java
package com.example.kr_kotlin.ui.registration_java;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import com.example.kr_kotlin.R;
import com.example.kr_kotlin.databinding.FragmentRegistrBinding;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ServerValue;

import java.util.HashMap;
import java.util.Map;

public class RegistrFragment extends Fragment {

    private FragmentRegistrBinding binding;

    @Nullable
    @Override
    public View onCreateView(
            @NonNull LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState
    ) {
        binding = FragmentRegistrBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(
            @NonNull View view,
            @Nullable Bundle savedInstanceState
    ) {
        super.onViewCreated(view, savedInstanceState);

        binding.buttonCreateAcc.setOnClickListener(v -> {
            String email = binding.emailEt.getText().toString().trim();
            String password = binding.passwordEt.getText().toString().trim();
            String repassword = binding.repasswordEt.getText().toString().trim();

            if (email.isEmpty() || password.isEmpty() || repassword.isEmpty()) {
                Toast.makeText(requireContext(),
                        "Поля не могут быть пустыми",
                        Toast.LENGTH_SHORT).show();
                return;
            }

            if (!password.equals(repassword)) {
                Toast.makeText(requireContext(),
                        "Пароли должны быть одинаковыми",
                        Toast.LENGTH_SHORT).show();
                return;
            }

            FirebaseAuth.getInstance()
                    .createUserWithEmailAndPassword(email, password)
                    .addOnCompleteListener(task -> {
                        if (task.isSuccessful()) {
                            saveEmailToDatabase(email, view);
                        } else {
                            Toast.makeText(requireContext(),
                                    "Ошибка регистрации: " +
                                            task.getException().getMessage(),
                                    Toast.LENGTH_SHORT).show();
                        }
                    });
        });
    }

    private void saveEmailToDatabase(String email, View view) {
        String userId = FirebaseAuth.getInstance().getCurrentUser() != null
                ? FirebaseAuth.getInstance().getCurrentUser().getUid()
                : null;

        if (userId == null) {
            Toast.makeText(requireContext(),
                    "Ошибка: пользователь не найден",
                    Toast.LENGTH_SHORT).show();
            return;
        }

        Map<String, Object> userData = new HashMap<>();
        userData.put("email", email);
        userData.put("createdAt", ServerValue.TIMESTAMP);

        FirebaseDatabase.getInstance().getReference()
                .child("Users")
                .child(userId)
                .setValue(userData)
                .addOnCompleteListener(dbTask -> {
                    if (dbTask.isSuccessful()) {
                        Navigation.findNavController(view)
                                .navigate(R.id.action_registrFragment_to_authFragment);
                    } else {
                        Toast.makeText(requireContext(),
                                "Ошибка сохранения данных: " +
                                        dbTask.getException().getMessage(),
                                Toast.LENGTH_SHORT).show();
                    }
                });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/kr_kotlin/ui/registration/RegistrFragment.kt
package com.example.kr_kotlin.ui.registration

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.navigation.fragment.findNavController
import com.example.kr_kotlin.R
import com.example.kr_kotlin.databinding.FragmentProfileBinding
import com.example.kr_kotlin.databinding.FragmentRegistrBinding
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ServerValue

class RegistrFragment : Fragment() {

    private var __binding : FragmentRegistrBinding? = null
    private val mBinding get() = __binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        __binding = FragmentRegistrBinding.inflate(layoutInflater, container, false)
        return mBinding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
//        val email = mBinding.emailEt.text.toString()
//        val password = mBinding.passwordEt.text.toString()
//        val repassword = mBinding.repasswordEt.text.toString()

        mBinding.buttonCreateAcc.setOnClickListener {
            val email = mBinding.emailEt.text.toString()
            val password = mBinding.passwordEt.text.toString()
            val repassword = mBinding.repasswordEt.text.toString()

            if (email.isEmpty() || password.isEmpty() || repassword.isEmpty()) {
                Toast.makeText(requireContext(),"Поля не могут быть пустыми", Toast.LENGTH_SHORT)
                    .show()
                return@setOnClickListener
            }
            else if (password != repassword) {
                Toast.makeText(requireContext(),"Пароли должны быть одинаковыми",
                    Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            else {
                FirebaseAuth.getInstance().createUserWithEmailAndPassword(email, password)
                    .addOnCompleteListener { task ->
                        if (task.isSuccessful) {
                            saveEmailToDataBase(email)
                        }
                        else {
                            Toast.makeText(requireContext(), "Ошибка регистрации: " +
                                    "${task.exception?.message}", Toast.LENGTH_SHORT).show()
                        }
                    }
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        __binding = null
    }

    private fun saveEmailToDataBase(email: String) {
        val userId = FirebaseAuth.getInstance().currentUser?.uid ?: run {
            Toast.makeText(requireContext(), "Ошибка: пользователь не найден",
                Toast.LENGTH_SHORT).show()
            return
        }
        val userData = hashMapOf("email" to email, "createdAt" to ServerValue.TIMESTAMP)

        FirebaseDatabase.getInstance().reference.child("Users").child(userId)
            .setValue(userData)
        findNavController().navigate(R.id.action_registrFragment_to_authFragment)
    }

}